lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,223 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  static get INCLUDE_INCOME_TRANSACTIONS() {
+    return 1 >> 1
+  }
+
+  static get INCLUDE_OUTCOME_TRANSACTIONS() {
+    return 1 >> 2
+  }
+
+  static get INCLUDE_ALL_TRANSACTIONS() {
+    CardanoAPI.INCLUDE_INCOME_TRANSACTIONS |
+      CardanoAPI.INCLUDE_OUTCOME_TRANSACTIONS
+  }
+
+  static get PER_PAGE_MAXIMUM() {
+    return 50
+  }
+
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, returnFullResponse = false) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return this.client({
+      method,
+      url,
+      params,
+      data
+    }).then(response => {
+      if (response.error) throw response.error
+
+      if (returnFullResponse) {
+        return response
+      }
+
+      return response.data.data
+    })
+  }
+
+  async fetchAllPages(endpoint, params, data) {
+    const fetchAllPagesAccumular = (page, dataAccumular = []) =>
+      this.fetch(endpoint, { ...params, page }, data, true).then(response => {
+        const responseData = Array.isArray(response.data.data)
+          ? response.data.data
+          : [response.data.data]
+        const result = [...responseData, ...dataAccumular]
+
+        if (
+          response.data.meta.pagination.totalPages >
```

Comment:
Is there a `totalItems` count in the response? If so, we should assert that it won't change during "iteration"

Reply:
and if it is changed, then what - start's again from the beginning? because this seems as only possible way for me

Reply:
Just fail the whole call. The caller should be able to deal with failures anyway